java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -285,30 +369,105 @@ private API getXPAPI(ExpressionTree et) {
       if (!mit.getMethodSelect().getKind().equals(Kind.MEMBER_SELECT)) {
         return API.UNKNOWN;
       }
+      if (mit.getArguments().size() <= 2) {
+        if (mit.getArguments().isEmpty()) {
+          if (xpFlagName.equals(EMPTY)) {
+            return getXPAPIFromMIT(mit);
+          } else {
+            return API.UNKNOWN;
+          }
+        } else {
+          ExpressionTree arg = mit.getArguments().get(0);
+          Symbol argSym = ASTHelpers.getSymbol(arg);
+          if (isLiteralTreeAndMatchesFlagName(arg)
+              || isVarSymbolAndMatchesFlagName(argSym)
+              || isSymbolAndMatchesFlagName(argSym)) {
+            return getXPAPIFromMIT(mit);
+          }
+        }
+      }
+    }
+    return API.UNKNOWN;
+  }
+
+  private API getXPAPIFromMIT(MethodInvocationTree mit) {
```

Comment:
I very well might be missing something, but this looks both like more code _and_ a worse runtime hit than just keeping the separate maps.

Reply:
See my comment above on the structure of the HashMap. This will simplify to one `containsKey` on `configMethodsMap`.

Reply:
That's not the current structure, though, check the four constants being looked up below.

Reply:
Agreed. let us revamp the map structure as discussed above. 

For a majority of the functions, the `containsKey` will be a false. For the few FF APIs that match, we can evaluate their attributes (methodtype, return type, receiver type, etc) and take appropriate action.